### ISSUE TITLE
User Id in instantiation data

### ISSUE THREAD

**Issue by srodriguezOPT** [sonata-nfv/son-gkeeper#697]
Hi @jbonnet ,

Besides the current service_uuid, ingress and egress fields, a new one will be sent to the gatekeeper in the instantiation request: the user_id. 

`{
       service_uuid: "8c58b169-7c38-4bcd-9421-a91bd786f1fa", 
       ingress: [], 
       egress: [], 
       user_id: "123-456"
}`

This is because the gatekeeper will need to check if the user has a valid license associated with that service before allowing him/her to instantiate it.

**Comment by jbonnet** [comment]
Hey, @srodriguezOPT  
If it is only the GK who needs the `user_id`, my suggestion is that it does not get passed to the GUI at all; what do you think?

**Comment by srodriguezOPT** [comment]
@jbonnet, 
yes, only the GK needs it but, how can the GK know what user made the request and if him/her has the associated license? by user_name?

**Comment by jbonnet** [comment]
@srodriguezOPT 
...by looking for the user name in the UM, using the `token` received in the request... am I missing something?
Then, if the service is private, by querying the Licence Store URL that came in the NSD...

**Comment by srodriguezOPT** [comment]
@jbonnet,

you're right. I was forgetting the token!
Ok, in that case, no user_id is needed.

**Comment by jbonnet** [comment]
@srodriguezOPT 
The GK can get the user from the token, as we do in other APIs... so the request would be something like:
```
$http.get(ENV.apiEndpoint+"/licenses")
``` 
with a valid `token` in the header. Would this fit the BSS's needs?

**Comment by srodriguezOPT** [comment]
@jbonnet 

Ok, great. 
So we do not need to include any user_id, it's enough with the given token.

get /licenses or /licences???

**Comment by jbonnet** [comment]
@srodriguezOPT 
:-)
I guess we're using **en-UK**, so it would be `GET .../licences` (beware though: the verb should be  **to license**)

**Comment by jbonnet** [comment]
This issue includes #603

**Comment by jbonnet** [comment]
@srodriguezOPT @tsoenen @DarioValocchi 
So...
The GK will start sending the SLM, besides the NSD/VNFDs and the service instance UUID (as the correlation ID), the following two arrays:
```
{
  "service_uuid": "8c58b169-7c38-4bcd-9421-a91bd786f1fa",
  "egresses": [
    {     
    }
  ],
  "ingresses": [
    {
      "location": "Aveiro",
      "nap": "123456789"
    },
    {
      "location": "London",
      "nap": "123123"
    }
  ]
```
@srodriguezOPT : please note the plural `ingresses`, to be coherent with `egresses`...

**Comment by tsoenen** [comment]
Ok, ingresses and egresses are both a list with dictionaries as elements. Both list will always have the same amount of elements?

**Comment by jbonnet** [comment]
@tsoenen no, I don't think both lists will always have the same number of elements.

**Comment by jbonnet** [comment]
@srodriguezOPT 
I think we had decided that there was no need for the GK to store egresses/ingresses with the request: can you please confirm that?

**Comment by srodriguezOPT** [comment]
@jbonnet,

ingresses and egresses: ok, both field names in plural.

Ok. both Gatekeeper neither BSS will not store the ingresses/egresses upon the request is sent

**Comment by jbonnet** [comment]
Closed by PR #759 
Might still need some polishing, after the BSS and the SLM do some more extensive tests
